Add: Mobile responsive Members page

willm30 opened the pull request:
## Description

This PR is for a mobile responsive Members page, per screen 5 of the [Figma design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide).

**New stuff** ✨

* Extracted `ColonyHomeInfo` from `ColonyHomeLayout` for reuse in `ColonyMembers`

**Changes** 🏗

*  `ColonyMembers`, `Members`, `MembersListItem` and related components styled for mobile.

## Screenshots

**_Members page_**
![mem-mob-page](https://user-images.githubusercontent.com/64402732/175572866-653aa192-1750-41cb-a311-283b4516217b.png)

**_Teams dropdown_**
![teams-menu](https://user-images.githubusercontent.com/64402732/175574000-c76c9d2b-985d-4cf2-822f-44517f4b4e8d.png)

**_Kebab_**
![kebab-mob](https://user-images.githubusercontent.com/64402732/175573731-0613eb68-91d1-45ac-96d2-8880f7494104.png)

**_Info popover_**
![info-popover-mob](https://user-images.githubusercontent.com/64402732/175573715-e98fbba7-c6fb-4eb3-b6c7-149a16c80735.png)

**_Pagination set to 10 members_**
![mem-mob-page](https://user-images.githubusercontent.com/64402732/175499549-23c2bc96-d1b4-480a-86a8-839449da6a1b.png)

Resolves  #3479

arrenv reviewed (changes requested):
Nice one @willm30.

Regarding the branch naming, it is good to have the issue you're resolving in the branch name as of the Git Workflow notion doc. For me at least, it helps with finding the right branch when checking it out. As of the screenshot.

![Screenshot 2022-06-24 110404](https://user-images.githubusercontent.com/33682027/175507873-6a012db7-4a9e-45d2-8fe6-eafa324150dc.png)

I think this PR needs to be rebased on the Contributors and Watchers branch, as there are a lot of changes to the members page on that branch.

Regarding this one.

There needs to be more margin on the right side of the kebab menu in each member item, you could match the left side. It will make it easier to touch on mobile screens.

![90408283-fad1-45e8-a02a-437b1d0eb7c9](https://user-images.githubusercontent.com/33682027/175508406-c1e431c7-e043-4afc-83e7-0093eb8dd0d3.png)

I like what you did with the permissions, however, this kebab menu has been added in the Contributors and Watchers branch, so, this is going to clash. I think permissions just needs to be hidden on mobile altogether, as permissions appears in the avatar popover anyway. Although, this is also broken.

![19c0daec-8033-49ec-8776-ec951dc2c876](https://user-images.githubusercontent.com/33682027/175508735-58bb5ecc-dd21-4536-987b-070b5cf96d01.png)

rdig reviewed (changes requested):
I found mostly what Arren found, with one extra, as the background for the Members page should be all white _(as in the specs)_. It will help ensure better readability.

Specs:
![Screenshot from 2022-06-24 16-36-52](https://user-images.githubusercontent.com/1193222/175547191-fd79677d-8f43-4af5-bc78-4d52bde4e710.png)

Current:
![Screenshot from 2022-06-24 15-54-37](https://user-images.githubusercontent.com/1193222/175547265-f870db1b-6619-40b8-947e-19a813c72eb3.png)

willm30 commented on src/modules/dashboard/components/Members/MembersTitle.css:
```diff
@@ -88,3 +90,32 @@
+    width: 100%;
+  }
+
+  .titleLeft ul[class*="SelectListBox"] {
+    width: 200px;
+    top: calc(100% + 2px);
+    right: 0px;
+  }
+
+  .titleLeft button[class*="CreateDomainButton"] {
+    justify-content: flex-start;
+    margin-left: 10px;
+    gap: 10px;
+  }
+
+  .titleLeft div[class*="CreateDomainButton"] {
+    position: static;
+  }
```
Styling the `ColonyDomainSelector`

willm30 commented on src/modules/dashboard/components/Members/MembersTitle.tsx:
```diff
@@ -78,66 +88,78 @@ const MembersTitle = ({
           text={MSG.title}
           appearance={{ size: 'medium', theme: 'dark' }}
+          textValues={{ isMobile }}
         />
```
All of the below is just adding the `ColonyDomainSelector` when  `isMobile` is true

willm30 commented:
@arrenv Apologies about the branch name, I will be conscious of this in future branches!

I've rebased the `feature/dapp-mobile-responsive` branch from `contributors-vs-watchers` so this branch is now in sync, and I've addressed the issues in your comment re the info popover. 

@rdig I also fixed the mobile colour scheme on `feature/dapp-mobile-responsive` so this page is now in white as it should be!

arrenv reviewed (changes requested):
It is looking better, nice one!

Just a couple of small things, the kebab icons could be a little bigger on mobile, and the menu from the kebab menu should have some padding.

![download (20)](https://user-images.githubusercontent.com/33682027/175904941-4fb2e967-2680-4dc7-8adf-f06ca44ec8c0.png)

willm30 commented:
@arrenv Padding added and kebab menu icon enlarged!

rdig reviewed (approved):
Good with this one as well. Nice job

![Screenshot from 2022-06-29 15-05-34](https://user-images.githubusercontent.com/1193222/176432182-d5635882-cbd9-4876-8ce2-8643043b927c.png)

arrenv reviewed (approved):
@willm30 looking good, I'm approving, although there is a little something weird with the dropdown padding. Particularly when logged in vs not being logged in, they are not the same. For example:

Logged in:

![download (22)](https://user-images.githubusercontent.com/33682027/176652925-99b0727d-62b7-40a4-9c34-45f573748a7e.png)


Not logged in:

![download (23)](https://user-images.githubusercontent.com/33682027/176652917-07a4c577-4dde-4a88-a15a-bb9e4cb6df93.png)

willm30 commented:
> @willm30 looking good, I'm approving, although there is a little something weird with the dropdown padding. Particularly when logged in vs not being logged in, they are not the same. For example:
> 

Great spot, I have fixed.
